src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -1715,7 +1715,36 @@ SPECTRE_ALWAYS_INLINE constexpr auto mutate_apply(
       "The stateless mutator does not specify both 'argument_tags' and "
       "'return_tags'. Did you forget to add these tag lists to the mutator "
       "class? The class is listed as the first template parameter below.");
-  mutate_apply(F{}, box, std::forward<Args>(args)...);
+  return mutate_apply(F{}, box, std::forward<Args>(args)...);
+}
+
+namespace DataBox_detail {
+template <typename... MutateTags, typename BoxTags, typename... Args>
+SPECTRE_ALWAYS_INLINE constexpr void mutate_assign_impl(
+    // NOLINTNEXTLINE(readability-avoid-const-params-in-decls)
+    const gsl::not_null<DataBox<BoxTags>*> box,
+    tmpl::list<MutateTags...> /*meta*/, Args&&... args) noexcept {
+  static_assert(sizeof...(MutateTags) == sizeof...(args),
+                "The number of arguments passed to `mutate_assign` must be "
+                "equal to the number of tags passed.");
+  db::mutate<MutateTags...>(box, [&args...](const auto... box_args) noexcept {
+    // silence unused capture warnings
```

Comment:
ping

Reply:
For compiler reasons I do not entirely understand, this warns even when I have ensured that nothing that calls it does so with 0 tags.

Reply:
oh, it's because it's still being instantiated... :( How about just adding to the comment `when there are zero args. This function still gets instantiated despite the static_assert in the parent function.`